### DISCUSSION TITLE
Create composite types

### DISCUSSION THREAD

**Discussion by PreetamSing** [SeaQL/sea-query#612] [Ideas]
What I'm trying to do:
```sql
CREATE TYPE my_composite_struct AS (code_type TEXT, reference_code INT);
CREATE TABLE demo_tb (id SERIAL PRIMARY KEY, composite_field my_composite_struct);
```
I couldn't find any way to do this in my migrations. What I need is a (rough) equivalent of how we create enum types in the following code, but instead for creating **composite** types:
```rust
use sea_orm_migration::prelude::*;
use sea_orm_migration::sea_query::extension::postgres::Type;

#[derive(Iden)]
pub enum KycStatus {
    #[iden = "kyc_status"]
    Type,
    Pending,
    Approved,
}

impl KycStatus {
    pub const fn list_variants() -> [KycStatus; 2] {
        [KycStatus::Pending, KycStatus::Approved]
    }
}

pub struct Migration;
// ...

#[async_trait::async_trait]
impl MigrationTrait for Migration {
    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
        let mut create_kyc_status_enum = Type::create();
        create_kyc_status_enum.as_enum(KycStatus::Type).values(KycStatus::list_variants());
        manager.create_type(create_kyc_status_enum).await?;
        // ...
        Ok(())
    }
    // ...
}
```

Digging a little bit, I found this commented variant of `TypeAs` enum:
https://github.com/SeaQL/sea-query/blob/3cb9247b2ab345d5795f84e696feee3842894779/src/extension/postgres/types.rs#L63
Let me know, if there are any existing solutions for this.

**Comment by tyt2y3** [comment]
There isn't support for composite type. Although as you have seen, there's a placeholder. But since Composite type is Postgres-only, you might as well just write raw SQL.
